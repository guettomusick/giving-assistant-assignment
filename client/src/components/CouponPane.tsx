import React from 'react';

const CouponPane = () => {
  return (
    <div>CouponPane</div>
  )
};

export default CouponPane;