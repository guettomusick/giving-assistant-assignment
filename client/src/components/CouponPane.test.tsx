import React from 'react';
import renderer from 'react-test-renderer';

import CouponPane, { Props } from './CouponPane';

describe('CouponPane', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      cmsData: {
        id: 'id',
        title: 'Coupon',
        description: 'A Coupon',
        code: 'CODE',
        active: true,
        cta: {
          title: 'Coupon',
        }
      }
    }
  })
  
  it('renders', () => {
    const component = renderer.create(
      <CouponPane { ...props } />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
