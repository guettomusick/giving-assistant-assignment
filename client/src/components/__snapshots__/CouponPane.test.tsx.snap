// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CouponPane renders 1`] = `
<div
  className="sc-AxgMl goCjCL"
>
  <div
    className="sc-AxheI fGpwZQ"
  >
    <h1
      className="sc-AxmLO YikNX"
    >
      Coupon
    </h1>
  </div>
  <div
    className="sc-fzozJi BUFJz"
  >
    <p>
      A Coupon
    </p>
  </div>
  <div
    className="sc-AxirZ eaZSdP"
  >
    <input
      className="sc-AxiKw lbWWnB"
      disabled={true}
      name="code"
      onChange={[Function]}
      value="CODE"
    />
    <button
      className="sc-AxjAm sc-AxhCb gLFHnk"
      onClick={[Function]}
    >
      Copy
    </button>
  </div>
</div>
`;
